### ISSUE TITLE
Add blog taxonomies

### ISSUE THREAD

**Issue by saschadube** [pagekit/pagekit#3]

**Comment by zhangfangsong** [comment]
now can I publish an article with a category ？

**Comment by Xarcell** [comment]
@zhangfangsong  No, there are still no tags or categories for blog posts.

**Comment by pseudomac** [comment]
**Xarcell**  I was looking for info about it, I hope that the management of the categories will be implemented.

**Comment by KevinSupertramp** [comment]
Hello,

Any news @ste93cry ? Is there a way we can help you working on it ? :-)

Best,

**Comment by ste93cry** [comment]
Sorry I didnt't have time to continue the work and I don't have anymore what I've done

**Comment by Wojozoo** [comment]
Any news regarding the blog categories?

**Comment by Nimothenicefish** [comment]
+1 for the blog categories

**Comment by KevinSupertramp** [comment]
Hi, is there anybody who's working on it ? If not I'll start to work on it.

**Comment by saschadube** [comment]
@KevinSupertramp No, we are not working on it at the moment. It would be great if you can help out :smile: Before you start coding, please post a detailed description how you want to implement this feature so we can discuss it.

**Comment by KevinSupertramp** [comment]
@saschadube Ok, so now I'm thinking really basic but at the same time it would be interresting to know where you want to go with taxonomies.

For now I thought about categories only but I don't know if you plan to implement tags and how (like if you want to implement tags the WordPress way then I would change the tables structures).

**Database side**:
Adding 2 tables:
* blog_category (id, slug, title, description and thumbnail)
* blog_post_categories (id, post_id, category_id) => 1 post can have more than 1 category

![pagekit - taxonomies - db](https://cloud.githubusercontent.com/assets/1142786/14111231/369e3ee0-f5ca-11e5-9885-a49cb3a42293.png)

**Admin side**:
Adding a tab after the tab "Posts" where you can manage categories (CRUD).
When adding or editing post I would add under "slug" or "status" an "intelligent search field" where you can define the post categories.

**Site side**:
Fetching posts by category.

--------------

I think that's it more or less, I really want to keep it simple because it's what I like in PageKit. Let me know if you have any suggestions or critics :wink:

**Comment by KevinSupertramp** [comment]
Any news @saschadube ? :)

**Comment by MalteScharenberg** [comment]
@KevinSupertramp, sorry for the long wait, we were really busy with the 1.0 release. 

I think a more generic approach would be great, so that tags and categories are possible. At pre alpha stages we've had a basic taxonomy implementation which tables were similar to this sketch: 

![tax_uml](https://cloud.githubusercontent.com/assets/3236302/14745260/cd13cf82-08aa-11e6-8b84-be919e94ebb7.png)
 
@malles already implemented categories for one of his Pagekit extensions (https://github.com/Bixie/pagekit-download), maybe this could be reused for a taxonomy implementation.

Help on this is really appreciated!

**Comment by ste93cry** [comment]
The approach described above is totally fine, it's how WordPress works and makes easy for new taxonomies to be added. However, there are also some disavantages: for example, what If I want to store more informations for a custom taxonomy (additional fields for example). WordPress has a term metadata table, but then there should be an extensible system too in Pagekit to register new taxonomies, etc.

**Comment by malles** [comment]
I have prepared a basic Taxonomy module to provide interfaces for extensions to use taxonomy in their items: https://github.com/pagekit/pagekit/compare/develop...malles:feature-taxonomy
This is pased on the Taxonomy that was implemented in the alpha version of Pagekit, with the current Node Models example.
@KevinSupertramp, @ste93cry or any above, please feel free to comment/contribute.

@ste93cry I think custom fields/settings can be stored in the data field of the Taxonomy or Term. If more is needed the extension should create his own tables and use the Traits.

In my view this module should provide the Traits and Models van maybe even user-interfaces for an extension to easily add taxonomy to their items. Blog would be the first place to experiment with that.

Please feel free to add suggestions to this issue!

**Comment by BelKoD** [comment]
Need use NestedSet in Doctrine!

**Comment by nikaro** [comment]
Any progress on this feature ? This is the only missing thing preventing me to switch to PageKit… :-)

**Comment by BelKoD** [comment]
I agree

**Comment by crwgregory** [comment]
One of the biggest frustrations of Wordpress taxonomies for me was that they could not have multiple parents.

**Comment by BelKoD** [comment]
For correct work sef need use one parent for first and many parents for second.

**Comment by amaierm** [comment]
Same here. I'd really appreciate to have that feature too. By the way, great job with pagekit so far. I'm lovin' it.

**Comment by AminulBD** [comment]
Is there any news?

**Comment by nickdring** [comment]
hi, any news on this? After the initial flood of releases since 1.00 its gonne pretty quiet.

**Comment by AminulBD** [comment]
@nickdring there is no news. I waited a long time for this, Now created an own system by forked (https://github.com/Bixie/pagekit-download) this.

**Comment by tanlinhnd** [comment]
Hi all, any news on this?

**Comment by Xarcell** [comment]
Looks like this feature has been abandoned. I wouldn't count on someone making this, simply because it's been attempted for years with no success. Someone will have to do it themselves or hire someone to code it in.

It's a shame though, no taxonomies has always been a deal breaker. I don't understand why the creators of pagekit didn't implement it,  within it's conception. Oh well, it's free, so you can't blame anyone. Although it's a shame though, pagekit had the potential to de-throne Wordpress, but with the lack of continued development, I don't see that happening.

**Comment by malles** [comment]
As a matter of fact, I am working on it this very moment ;)

**Comment by malles** [comment]
My next attempt to this: https://github.com/Bixie/pagekit-taxonomy
This can be used as an extension for now, do run webpack if you use it. Any feedback or ideas are welcome.

**Comment by asharamseervi** [comment]
Hello @pagekit contributors and community !!
Congratulations for Pagekit. Recently working with Pagekit and obviously loving it.
**I am also raising hand towards "Blog Taxonomy"**. This will help us to organize our posts to visitors. I had noticed, _this feature is requested around March, 2014 and still waiting for any resolutions_. Please enhance productivity and help this cms more easier and awesome.

**Comment by d9media** [comment]
I've tested malle's repository. Thank you for bringing this feature forward! I wasn't able to test it though, it installs but I was unable to activate it - it just doesn't show up as an extension. (I've commented on the other repository already)

I hate to bring this up here but I'm starting to get the same feeling as Xarcell. I was really excited about the project, but the lack of overall progress shows that the Dev's are not able to fully commit to making Pagekit great for end-users. It's becoming particulary obvious at this issue. God knows how many users were put of due to the lack of this feature. It's a total deal breaker for a CMS like pagekit, which is openly marketed towards the typical Wordpress users (small sites with just a few content types and some media). 

I know it's harsh to say that, but after nearly a year of testing I'm kinda giving up on Pagekit. I can only feel comfortable building my sites on a CMS that I can trust will be extended and supported long term. It's taken years to release the current version in the first place, how am I supposed to trust that it won't take another few years until it's useful from an end user perspective? It's obviously a great framework for extending and I'm sure it's very helpful for their own work, but from and end-users point of view, it won't work until key features are implemented. I don't blame anyone, but it's a shame really. So much potential.

**Comment by nickdring** [comment]
Well said d9media. Speaking from my POV we have and are building quite a few Pagekit sites. We'll probably end up with around 50+. We've developed our own extensions and we as developers like page kit and the users like the interface, especially those who are used to Joomla! BUT, I find it a little embarrassing when I have to explain why people can only use the blog for one category. We can get round it by building the pages by hand but thats not what a CMS is about.
I'm sure Sascha and the other guys at Yootheme are very busy what with Yootheme Pro and the other things but if you are going to launch a CMS you need to have a road plan of where things are going so people can see whats ahead and have confidence going forward.
I'd like to see a post from Sascha outlining where they are and whats to come in future releases with timings. We all work in Agile (don't we) so I'm sure things are fluid and I personally won't complain if feature x isn't in release x. A lot of us come from the Joomla world so we know all about releases slipping. :)

**Comment by ahmadsyamim** [comment]
Hi all!
Im also recently working with Pagekit and loving it.
However all the key features are not yet implemented. 
Im hoping for this cms to grow with success and awesome.

**Comment by nikaro** [comment]
This issue is more than 3 years old, i think you should forget about this feature unless you code it yourself… :-/

**Comment by csaeum** [comment]
+1

**Comment by AminulBD** [comment]
@csaeum Everyone wants that. But behind the YOOtheme team is maybe busy for years or don't want to develop this feature actually.
